-- ----------------------------------------------------------------------
-- SQL data bulk transfer script generated by the MySQL Migration Toolkit
-- ----------------------------------------------------------------------

-- Disable foreign key checks
SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;


INSERT INTO `gnomex`.`AnalysisType`(`idAnalysisType`, `analysisType`, `isActive`, `idAppUser`)
VALUES (1, 'ChIP-Seq analysis', 'Y', NULL),
  (2, 'TiMAT2 Tiling Analysis', 'Y', 566),
  (3, 'RNA-Seq, Differential', 'Y', 566),
  (4, 'Viral sequence screen', 'Y', 273),
  (5, 'small RNA-Seq', 'Y', 566),
  (6, 'aCGH', 'Y', 566),
  (7, 'Digital Gene Expression (DiGE)', 'Y', 566),
  (8, 'Microarray Design', 'Y', 566),
  (9, 'Alignment', 'Y', 566),
  (10, 'Cap-Seq', 'Y', 566),
  (11, 'SNP/INDEL', 'Y', 566);

INSERT INTO `gnomex`.`AppUser`(`idAppUser`, `lastName`, `firstName`, `uNID`, `email`, `phone`, `department`, `institute`, `jobTitle`, `isActive`, `codeUserPermissionKind`, `userNameExternal`, `passwordExternal`)
VALUES (1, '', 'admin', null, null, null, NULL, NULL, NULL, 'Y', 'ADMIN', 'admin', '0DPiKuNIrrVmD8IUCuw1hQxNqZc=');

INSERT INTO `gnomex`.`ArrayCoordinate`(`idArrayCoordinate`, `name`, `x`, `y`, `idSlideDesign`)
VALUES  (2,'1_1',1,1,54),
 (3,'1_2',1,2,54),
 (16,'1',1,1,3),
 (17,'1',1,1,4),
 (18,'1',1,1,5),
 (19,'1',1,1,6),
 (21,'1',1,1,8),
 (24,'1',1,1,11),
 (25,'1',1,1,12),
 (27,'1',1,1,14),
 (31,'1',1,1,18),
 (38,'1',1,1,25),
 (44,'1',1,1,31),
 (48,'1',1,1,36),
 (72,'1',1,1,115),
 (74,'1',1,1,118),
 (75,'1_3',1,3,128),
 (76,'1_1',1,1,128),
 (77,'1_4',1,4,128),
 (78,'1_2',1,2,128),
 (79,'1_3',1,3,17),
 (80,'1_1',1,1,17),
 (81,'1_4',1,4,17),
 (82,'1_2',1,2,17),
 (85,'1_3',1,3,131),
 (86,'1_1',1,1,131),
 (87,'1_4',1,4,131),
 (88,'1_2',1,2,131),
 (97,'1_3',1,3,15),
 (98,'1_1',1,1,15),
 (99,'1_4',1,4,15),
 (100,'1_2',1,2,15),
 (102,'1_1',1,1,32),
 (103,'1_2',1,2,32),
 (104,'1_3',1,3,54),
 (105,'1_4',1,4,54),
 (106,'1_3',1,3,26),
 (107,'1_1',1,1,26),
 (108,'1_4',1,4,26),
 (109,'1_2',1,2,26),
 (110,'1_3',1,3,132),
 (111,'1_1',1,1,132),
 (112,'1_4',1,4,132),
 (113,'1_2',1,2,132),
 (114,'1_3',1,3,133),
 (115,'1_1',1,1,133),
 (116,'1_4',1,4,133),
 (117,'1_2',1,2,133),
 (122,'1_3',1,3,135),
 (123,'1_1',1,1,135),
 (124,'1_4',1,4,135),
 (125,'1_2',1,2,135),
 (126,'1_3',1,3,136),
 (127,'1_1',1,1,136),
 (128,'1_4',1,4,136),
 (129,'1_2',1,2,136),
 (130,'1_1',1,1,117),
 (131,'1_2',1,2,117),
 (132,'1_3',1,3,139),
 (133,'1_1',1,1,139),
 (134,'1_4',1,4,139),
 (135,'1_2',1,2,139),
 (140,'1_3',1,3,33),
 (141,'1_1',1,1,33),
 (142,'1_4',1,4,33),
 (143,'1_2',1,2,33),
 (157,'1_3',1,3,153),
 (158,'1_4',1,4,153),
 (159,'1_1',1,1,153),
 (160,'1_2',1,2,153),
 (177,'1_3',1,3,164),
 (178,'1_4',1,4,164),
 (179,'1_1',1,1,164),
 (180,'1_2',1,2,164),
 (182,'2_4',2,4,167),
 (183,'2_1',2,1,167),
 (184,'1_3',1,3,167),
 (185,'2_3',2,3,167),
 (186,'1_4',1,4,167),
 (187,'1_1',1,1,167),
 (188,'2_2',2,2,167),
 (189,'1_2',1,2,167),
 (190,'2_4',2,4,165),
 (191,'2_1',2,1,165),
 (192,'1_3',1,3,165),
 (193,'2_3',2,3,165),
 (194,'1_4',1,4,165),
 (195,'1_1',1,1,165),
 (196,'2_2',2,2,165),
 (197,'1_2',1,2,165),
 (198,'1_1',1,1,168),
 (199,'1_3',1,3,168),
 (200,'1_4',1,4,168),
 (201,'1_2',1,2,168),
 (208,'1_1',1,1,175),
 (209,'1_3',1,3,175),
 (210,'1_4',1,4,175),
 (211,'1_2',1,2,175),
 (212,'1_1',1,1,177),
 (213,'2_1',2,1,177),
 (214,'1_3',1,3,177),
 (215,'1_4',1,4,177),
 (216,'1_2',1,2,177),
 (217,'2_2',2,2,177),
 (218,'2_4',2,4,177),
 (219,'2_3',2,3,177);



INSERT INTO `gnomex`.`BillingCategory`(`idBillingCategory`, `description`, `pluginClassName`, `codeBillingChargeKind`, `isActive`)
VALUES (1, 'Sample Quality', 'hci.gnomex.billing.SampleQualityPlugin', 'SERVICE', 'Y'),
  (2, 'Microarray Labeling Reaction', 'hci.gnomex.billing.LabelingReactionPlugin', 'SERVICE', 'Y'),
  (3, 'Microarray Analysis', 'hci.gnomex.billing.MicroarrayAnalysisPlugin', 'SERVICE', 'Y'),
  (4, 'Microarray', 'hci.gnomex.billing.MicroarrayPlugin', 'PRODUCT', 'Y'),
  (5, 'Illumina Library Preparation', 'hci.gnomex.billing.IlluminaLibPrepPlugin', 'SERVICE', 'Y'),
  (7, 'Illumina GA Sequencing', 'hci.gnomex.billing.IlluminaSeqPlugin', 'SERVICE', 'Y'),
  (8, 'Sample Quality (Illumina Lib)', 'hci.gnomex.billing.IlluminaLibSampleQualityPlugin', 'SERVICE', 'Y'),
  (9, 'Miscellaneous', NULL, 'SERVICE', 'Y'),
  (10, 'Array Capture', NULL, 'SERVICE', 'Y');

INSERT INTO `gnomex`.`BillingChargeKind`(`codeBillingChargeKind`, `billingChargeKind`, `isActive`)
VALUES ('PRODUCT', 'Product', 'Y'),
  ('SERVICE', 'Service', 'Y');


INSERT INTO `gnomex`.`BillingPeriod`(`idBillingPeriod`, `billingPeriod`, `startDate`, `endDate`, `isActive`)
VALUES (1, 'Jan 2009', '2009-01-01 00:00:00', '2009-01-31 00:00:00', 'Y'),
  (2, 'Feb 2009', '2009-02-01 00:00:00', '2009-02-28 00:00:00', 'Y'),
  (3, 'Mar 2009', '2009-03-01 00:00:00', '2009-03-31 00:00:00', 'Y'),
  (4, 'Apr 2009', '2009-04-01 00:00:00', '2009-04-30 00:00:00', 'Y'),
  (5, 'May 2009', '2009-05-01 00:00:00', '2009-05-31 00:00:00', 'Y'),
  (6, 'Jun 2009', '2009-06-01 00:00:00', '2009-06-30 00:00:00', 'Y'),
  (7, 'Jul 2009', '2009-07-01 00:00:00', '2009-07-31 00:00:00', 'Y'),
  (8, 'Aug 2009', '2009-08-01 00:00:00', '2009-08-31 00:00:00', 'Y'),
  (9, 'Sep 2009', '2009-09-01 00:00:00', '2009-09-30 00:00:00', 'Y'),
  (10, 'Oct 2009', '2009-10-01 00:00:00', '2009-10-31 00:00:00', 'Y'),
  (11, 'Nov 2009', '2009-11-01 00:00:00', '2009-11-30 00:00:00', 'Y'),
  (12, 'Dec 2009', '2009-12-01 00:00:00', '2009-12-31 00:00:00', 'Y'),
  (13, 'Jan 2010', '2010-01-01 00:00:00', '2010-01-31 00:00:00', 'Y'),
  (14, 'Feb 2010', '2010-02-01 00:00:00', '2010-02-28 00:00:00', 'Y'),
  (15, 'Mar 2010', '2010-03-01 00:00:00', '2010-03-31 00:00:00', 'Y'),
  (16, 'Apr 2010', '2010-04-01 00:00:00', '2010-04-30 00:00:00', 'Y'),
  (17, 'May 2010', '2010-05-01 00:00:00', '2010-05-31 00:00:00', 'Y'),
  (18, 'Jun 2010', '2010-06-01 00:00:00', '2010-06-30 00:00:00', 'Y'),
  (19, 'Jul 2010', '2010-07-01 00:00:00', '2010-07-31 00:00:00', 'Y'),
  (20, 'Aug 2010', '2010-08-01 00:00:00', '2010-08-31 00:00:00', 'Y'),
  (21, 'Sep 2010', '2010-09-01 00:00:00', '2010-09-30 00:00:00', 'Y'),
  (22, 'Oct 2010', '2010-10-01 00:00:00', '2010-10-31 00:00:00', 'Y'),
  (23, 'Nov 2010', '2010-11-01 00:00:00', '2010-11-30 00:00:00', 'Y'),
  (24, 'Dec 2010', '2010-12-01 00:00:00', '2010-12-31 00:00:00', 'Y'),
  (25, 'Jan 2011', '2011-01-01 00:00:00', '2011-01-31 00:00:00', 'Y'),
  (26, 'Feb 2011', '2011-02-01 00:00:00', '2011-02-28 00:00:00', 'Y'),
  (27, 'Mar 2011', '2011-03-01 00:00:00', '2011-03-31 00:00:00', 'Y'),
  (28, 'Apr 2011', '2011-04-01 00:00:00', '2011-04-30 00:00:00', 'Y'),
  (29, 'May 2011', '2011-05-01 00:00:00', '2011-05-31 00:00:00', 'Y'),
  (30, 'Jun 2011', '2011-06-01 00:00:00', '2011-06-30 00:00:00', 'Y'),
  (31, 'Jul 2011', '2011-07-01 00:00:00', '2011-07-31 00:00:00', 'Y'),
  (32, 'Aug 2011', '2011-08-01 00:00:00', '2011-08-31 00:00:00', 'Y'),
  (33, 'Sep 2011', '2011-09-01 00:00:00', '2011-09-30 00:00:00', 'Y'),
  (34, 'Oct 2011', '2011-10-01 00:00:00', '2011-10-31 00:00:00', 'Y'),
  (35, 'Nov 2011', '2011-11-01 00:00:00', '2011-11-30 00:00:00', 'Y'),
  (36, 'Dec 2011', '2011-12-01 00:00:00', '2011-12-31 00:00:00', 'Y'),
  (37, 'Dec 2008', '2008-12-01 00:00:00', '2008-12-31 00:00:00', 'Y'),
  (38, 'Nov 2008', '2008-11-01 00:00:00', '2008-11-30 00:00:00', 'Y'),
  (39, 'Oct 2008', '2008-10-01 00:00:00', '2008-10-31 00:00:00', 'Y'),
  (40, 'Sep 2008', '2008-09-01 00:00:00', '2008-09-30 00:00:00', 'Y');

INSERT INTO `gnomex`.`BillingPrice`(`idBillingPrice`, `idBillingCategory`, `idBillingTemplate`, `description`, `filter1`, `filter2`, `unitPrice`, `isActive`)
VALUES  (1,1,NULL,'Bioanalyzer RNA Nano','RNANANO',NULL,'0.00','Y'),
 (2,1,NULL,'Bioanalyzer RNA Pico','RNAPICO',NULL,'0.00','Y'),
 (3,1,NULL,'Bioanalyzer DNA 1000','DNA1000',NULL,'0.00','Y'),
 (4,1,NULL,'Bioanalyzer Small RNA','SMALLRNA',NULL,'0.00','Y'),
 (5,2,2,'Agilent Gene Expression','AGIL','EXP','0.00','Y'),
 (6,2,2,'Agilent CGH','AGIL','CGH','0.00','Y'),
 (7,2,2,'Agilent ChIP-on-chip','AGIL','CHIP','0.00','Y'),
 (8,2,2,'Agilent DNA Methylation','AGIL','METH','0.00','Y'),
 (9,2,2,'Agilent HybMap','AGIL','WTRANSCRP','0.00','Y'),
 (10,2,2,'Agilent SNP','AGIL','SNP','0.00','Y'),
 (11,2,2,'Agilent miRNA','AGIL','MIRNA','0.00','Y'),
 (12,2,2,'Agilent Exon','AGIL','EXON','0.00','Y'),
 (13,3,2,'Agilent Gene Expression 44K','1',NULL,'0.00','Y'),
 (14,3,2,'Aglient CGH 244K','2',NULL,'0.00','Y'),
 (15,3,2,'Agilent CGH 105K','3',NULL,'0.00','Y'),
 (16,3,2,'Agilent CGH 44K','4',NULL,'0.00','Y'),
 (17,3,2,'Agilent CGH 15K','5',NULL,'0.00','Y'),
 (18,3,2,'Agilent ChIP-on-chip 244K','6',NULL,'0.00','Y'),
 (19,3,2,'Agilent ChIP-on-chip 105K','7',NULL,'0.00','Y'),
 (20,3,2,'Agilent ChIP-on-chip 44K','8',NULL,'0.00','Y'),
 (21,3,2,'Agilent ChIP-on-chip 15K','9',NULL,'0.00','Y'),
 (22,3,2,'Agilent HybMap 244K','10',NULL,'0.00','Y'),
 (23,3,2,'Agilent HybMap 44K','11',NULL,'0.00','Y'),
 (24,3,3,'Affymetrix Gene Expression','12',NULL,'0.00','Y'),
 (25,3,3,'Affymetrix SNP','13',NULL,'0.00','Y'),
 (26,4,2,'Agilent 244K Microarray','1',NULL,'0.00','Y'),
 (27,4,2,'Agilent 105K Microarray','2',NULL,'0.00','Y'),
 (28,4,2,'Agilent 44K Microarray','3',NULL,'0.00','Y'),
 (29,4,2,'Agilent 15K Microarray','4',NULL,'0.00','Y'),
 (30,4,3,'Affymetrix SNP 250K Microarray','5',NULL,'0.00','Y'),
 (31,5,NULL,'Genomic DNA','1',NULL,'0.00','Y'),
 (32,5,NULL,'ChIP-Seq','7',NULL,'0.00','Y'),
 (33,5,NULL,'Small RNA','12',NULL,'0.00','Y'),
 (34,5,NULL,'mRNA-Seq','4',NULL,'0.00','Y'),
 (35,7,NULL,'18 Cycle SR','1','3','0.00','Y'),
 (36,7,NULL,'26 Cycle SR','2','3','0.00','Y'),
 (37,7,NULL,'36 Cycle SR','3','3','0.00','Y'),
 (38,7,NULL,'18 Cycle PE','1','4','0.00','Y'),
 (39,7,NULL,'26 Cycle PE','2','4','0.00','Y'),
 (40,7,NULL,'36 Cycle PE','3','4','0.00','Y'),
 (41,2,3,'Affymetrix Gene Expression Labeling','AFFY','12','0.00','Y'),
 (42,2,3,'Affymetrix 250K SNP Labeling','AFFY','13','0.00','Y'),
 (43,1,NULL,'Qubit PicoGreen Assay','PICOGREEN',NULL,'0.00','Y'),
 (44,1,NULL,'gDNA gel','DNAGEL',NULL,'0.00','Y'),
 (45,3,2,'Agilent HybMap 15K','15',NULL,'0.00','Y'),
 (46,9,2,'Slide Re-use Preparation',NULL,NULL,'0.00','Y'),
 (47,8,NULL,'Bioanalyzer RNA Nano','RNANANO',NULL,'0.00','Y'),
 (48,8,NULL,'Bioanalyzer RNA Pico','RNAPICO',NULL,'0.00','Y'),
 (49,8,NULL,'Bioanalyzer DNA 1000','DNA1000',NULL,'0.00','Y'),
 (50,8,NULL,'Bioanalyzer Small RNA','SMALLRNA',NULL,'0.00','Y'),
 (51,8,NULL,'Qubit PicoGreen Assay','PICOGREEN',NULL,'0.00','Y'),
 (52,8,NULL,'gDNA gel','DNAGEL',NULL,'0.00','Y'),
 (53,3,2,'Agilent Zebrafish Custom','16',NULL,'0.00','Y'),
 (54,3,2,'Agilent 15K miRNA','16',NULL,'0.00','Y'),
 (55,2,3,'Affymetrix 5.0/6.0 SNP Labeling','AFFY','14','0.00','Y'),
 (57,4,3,'Affymetrix SNP 5.0 Microarray','6',NULL,'0.00','Y'),
 (58,4,3,'Affymetrix Gene Expression Microarray','7',NULL,'0.00','Y'),
 (59,2,2,'Agilent cDNA synthesis/Agilent ChIP-on-chip Labeling',NULL,NULL,'0.00','Y'),
 (60,5,NULL,'Total RNA (eukaryote)','5',NULL,'0.00','Y'),
 (61,5,NULL,'Total RNA (prokaryote)','9',NULL,'0.00','Y'),
 (62,5,NULL,'cDNA','10',NULL,'0.00','Y'),
 (63,5,NULL,'mRNA-Seq','11',NULL,'0.00','Y'),
 (64,3,2,'Agilent Gene Expression 244K','17',NULL,'0.00','Y'),
 (65,4,3,'Affymetrix SNP 6.0 Microarray','8',NULL,'0.00','Y'),
 (66,4,3,'Affymetrix Gene Expression (rat) Microarray','9',NULL,'0.00','Y'),
 (67,4,3,'Affymetrix Gene Expression (zebrafish) Microarray','10',NULL,'0.00','Y'),
 (68,4,2,'Agilent miRNA 15K','11',NULL,'0.00','Y'),
 (69,4,2,'Agilent 1M microarray','12',NULL,'0.00','Y'),
 (70,3,2,'Agilent HybMap 1M','18',NULL,'0.00','Y'),
 (71,2,2,'Agilent HQ Gene Expression',NULL,NULL,'0.00','Y'),
 (72,10,NULL,'Agilent 1X244K',NULL,NULL,'0.00','Y'),
 (73,10,NULL,'Agilent 2x105K',NULL,NULL,'0.00','Y'),
 (74,10,NULL,'Agilent 4x44K',NULL,NULL,'0.00','Y'),
 (75,4,2,'Agilent 180K microarray','19',NULL,'0.00','Y'),
 (76,3,2,'Agilent CGH 180K','13',NULL,'0.00','Y'),
 (77,7,NULL,'76 Cycle SR','4','3','0.00','Y'),
 (78,7,NULL,'76 Cycle PE','4','4','0.00','Y'),
 (85,2,5,'Agilent Gene Expression','AGIL1','EXP','0.00','Y'),
 (86,2,5,'Agilent CGH','AGIL1','CGH','0.00','Y'),
 (87,2,5,'Agilent ChIP-on-chip','AGIL1','CHIP','0.00','Y'),
 (88,2,5,'Agilent DNA Methylation','AGIL1','METH','0.00','Y'),
 (89,2,5,'Agilent HybMap','AGIL1','WTRANSCRP','0.00','Y'),
 (110,2,5,'Agilent SNP','AGIL1','SNP','0.00','Y'),
 (111,2,5,'Agilent miRNA','AGIL1','MIRNA','0.00','Y'),
 (112,2,5,'Agilent Exon','AGIL1','EXON','0.00','Y'),
 (113,3,5,'Agilent Gene Expression 44K','1',NULL,'0.00','Y'),
 (114,3,5,'Aglient CGH 244K','2',NULL,'0.00','Y'),
 (115,3,5,'Agilent CGH 105K','3',NULL,'0.00','Y'),
 (116,3,5,'Agilent CGH 44K','4',NULL,'0.00','Y'),
 (117,3,5,'Agilent CGH 15K','5',NULL,'0.00','Y'),
 (118,3,5,'Agilent ChIP-on-chip 244K','6',NULL,'0.00','Y'),
 (119,3,5,'Agilent ChIP-on-chip 105K','7',NULL,'0.00','Y'),
 (120,3,5,'Agilent ChIP-on-chip 44K','8',NULL,'0.00','Y'),
 (121,3,5,'Agilent ChIP-on-chip 15K','9',NULL,'0.00','Y'),
 (122,3,5,'Agilent HybMap 244K','10',NULL,'0.00','Y'),
 (123,3,5,'Agilent HybMap 44K','11',NULL,'0.00','Y'),
 (126,4,5,'Agilent 244K Microarray','1',NULL,'0.00','Y'),
 (127,4,5,'Agilent 105K Microarray','2',NULL,'0.00','Y'),
 (128,4,5,'Agilent 44K Microarray','3',NULL,'0.00','Y'),
 (129,4,5,'Agilent 15K Microarray','4',NULL,'0.00','Y'),
 (130,4,5,'Affymetrix SNP 250K Microarray','5',NULL,'0.00','Y'),
 (145,3,5,'Agilent HybMap 15K','15',NULL,'0.00','Y'),
 (146,9,5,'Slide Re-use Preparation',NULL,NULL,'0.00','Y'),
 (153,3,5,'Agilent Zebrafish Custom','16',NULL,'0.00','Y'),
 (154,3,5,'Agilent 15K miRNA','16',NULL,'0.00','Y'),
 (159,2,5,'Agilent cDNA synthesis/Agilent ChIP-on-chip Labeling',NULL,NULL,'0.00','Y'),
 (164,3,5,'Agilent Gene Expression 244K','17',NULL,'0.00','Y'),
 (168,4,5,'Agilent miRNA 15K','11',NULL,'0.00','Y'),
 (169,4,5,'Agilent 1M microarray','12',NULL,'0.00','Y'),
 (170,3,5,'Agilent HybMap 1M','18',NULL,'0.00','Y'),
 (171,2,5,'Agilent HQ Gene Expression',NULL,NULL,'0.00','Y'),
 (175,4,5,'Agilent 180K microarray','19',NULL,'0.00','Y'),
 (176,3,5,'Agilent CGH 180K','13',NULL,'0.00','Y');


INSERT INTO `gnomex`.`BillingSlideProductClass`(`idBillingSlideProductClass`, `billingSlideProductClass`, `isActive`)
VALUES (1, 'Agilent 244K microarray', 'Y'),
  (2, 'Agilent 105K microarray', 'Y'),
  (3, 'Agilent 44K microarray', 'Y'),
  (4, 'Agilent 15K microarray', 'Y'),
  (5, 'Affymetrix 250K SNP microarray', 'Y'),
  (6, 'Affymetrix 5.0 SNP microarray', 'Y'),
  (7, 'Affymetrix Gene Expression microarray', 'Y'),
  (8, 'Affymetrix 6.0 SNP microarray', 'Y'),
  (9, 'Affymetrix Gene Expression (rat) microarray', 'Y'),
  (10, 'Affymetrix Gene Expression (zebrafish) microarray', 'Y'),
  (11, 'Agilent miRNA 15K microarray', 'Y'),
  (12, 'Agilent 1M microarray', 'Y'),
  (13, 'Agilent 180K microarray', 'Y');

INSERT INTO `gnomex`.`BillingSlideServiceClass`(`idBillingSlideServiceClass`, `billingSlideServiceClass`, `isActive`)
VALUES (1, 'Agilent Gene Expression 44K', 'Y'),
  (2, 'Aglient CGH 244K', 'Y'),
  (3, 'Agilent CGH 105K', 'Y'),
  (4, 'Agilent CGH 44K', 'Y'),
  (5, 'Agilent CGH 15K', 'Y'),
  (6, 'Agilent ChIP-on-chip 244K', 'Y'),
  (7, 'Agilent ChIP-on-chip 105K', 'Y'),
  (8, 'Agilent ChIP-on-chip 44K', 'Y'),
  (9, 'Agilent ChIP-on-chip 15K', 'Y'),
  (10, 'Agilent HybMap 244K', 'Y'),
  (11, 'Agilent HybMap 44K', 'Y'),
  (12, 'Affymetrix Gene Expression', 'Y'),
  (13, 'Affymetrix SNP 250K', 'Y'),
  (14, 'Affymetrix SNP 5.0/6.0', 'Y'),
  (15, 'Agilent Zebrafish Custom', 'Y'),
  (16, 'Agilent 15K miRNA', 'Y'),
  (17, 'Agilent Gene Expression 244K', 'Y'),
  (18, 'Agilent HybMap 1M', 'Y'),
  (19, 'Agilent CGH 180K', 'Y');

INSERT INTO `gnomex`.`BillingStatus`(`codeBillingStatus`, `billingStatus`, `isActive`)
VALUES ('APPROVED', 'Approved', 'Y'),
  ('APPROVEDEX', 'Approved (External)', 'Y'),
  ('COMPLETE', 'Completed', 'Y'),
  ('NEW', 'New', 'Y'),
  ('PENDING', 'Pending', 'Y');

INSERT INTO `gnomex`.`BillingTemplate`(`idBillingTemplate`, `description`, `codeRequestCategory`, `isActive`)
VALUES (1, 'Sample Quality Request', 'QC', 'Y'),
  (2, 'Agilent 2-color Microarray', 'AGIL', 'Y'),
  (3, 'Affymetrix Microarray', 'AFFY', 'Y'),
  (4, 'Illumina Sequencing', 'SOLEXA', 'Y'),
  (5, 'Agilent 1-color Microarray', 'AGIL1', 'Y');

INSERT INTO `gnomex`.`BillingTemplateEntry`(`idBillingTemplate`, `idBillingCategory`, `sortOrder`)
VALUES (1, 1, 1),
  (2, 1, 1),
  (3, 1, 1),
  (4, 1, 1),
  (2, 2, 2),
  (3, 2, 2),
  (2, 3, 3),
  (3, 3, 3),
  (2, 4, 4),
  (3, 4, 4),
  (4, 5, 2),
  (4, 7, 5),
  (4, 8, 3),
  (2, 9, 5),
  (4, 10, 6),
  (5, 1, 1),
  (5, 2, 2),
  (5, 3, 3),
  (5, 4, 4),
  (5, 9, 5);


INSERT INTO `gnomex`.`BioanalyzerChipType`(`codeBioanalyzerChipType`, `bioanalyzerChipType`, `concentrationRange`, `maxSampleBufferStrength`, `costPerSample`, `sampleWellsPerChip`, `isActive`, `codeConcentrationUnit`)
VALUES ('DNA1000', 'DNA 1000 Chip', '5 to 50 ng/µl', '250 mM KCl/NaCl', 10.00, 12, 'Y', 'ng/ul'),
  ('RNANANO', 'RNA Nano Chip', '25 to 500 ng/µl', '10mM TE', 7.00, 12, 'Y', 'ng/ul'),
  ('RNAPICO', 'RNA Pico Chip', '200 to 5000 pg/µl', '10 mM TE', 8.00, 11, 'Y', 'pg/ul'),
  ('SMALLRNA', 'Small RNA Chip', '1-20 ng/µl', NULL, NULL, 11, 'Y', 'pg/ul');

INSERT INTO `gnomex`.`ConcentrationUnit`(`codeConcentrationUnit`, `concentrationUnit`, `mageOntologyCode`, `mageOntologyDefinition`, `isActive`)
VALUES ('ng/ul', 'ng/ul', NULL, NULL, 'Y'),
  ('pg/ul', 'pg/ul', NULL, NULL, 'Y');

INSERT INTO `gnomex`.`ExperimentDesign`(`codeExperimentDesign`, `experimentDesign`, `mageOntologyCode`, `mageOntologyDefinition`, `isActive`, `idAppUser`)
VALUES ('BINDSITE', 'Binding Site Identification', 'binding_site_identification_design', NULL, 'Y', NULL),
  ('CGH', 'Comparative Genomic Hybridization', 'comparative_genome_hybridization_design', NULL, 'Y', NULL),
  ('CHROM', 'Chromatin Marks / Modifications', NULL, NULL, 'Y', NULL),
  ('DIFFEXPR', 'Differential Expression', NULL, NULL, 'Y', NULL),
  ('DIFFSPLICE', 'Differential Splicing', NULL, NULL, 'Y', NULL),
  ('EXPR', 'Expression', 'is_espressed_design', NULL, 'Y', NULL),
  ('OTHER', 'Other', NULL, NULL, 'Y', NULL),
  ('SEQENRICH', 'Sequence Enrichment', NULL, NULL, 'Y', NULL),
  ('TRANSCRIPT', 'Transcript Identification', 'transcript_identification_design', NULL, 'Y', NULL);


INSERT INTO `gnomex`.`ExperimentFactor`(`codeExperimentFactor`, `experimentFactor`, `mageOntologyCode`, `mageOntologyDefinition`, `isActive`, `idAppUser`)
VALUES ('AGE', 'Age / Developmental Stage', 'age', NULL, 'Y', NULL),
  ('CELLLINE', 'Cell Line / Strain', 'cell_line', NULL, 'Y', NULL),
  ('CELLTYPE', 'Cell Type', 'cell_type', NULL, 'Y', NULL),
  ('CLININFO', 'Clinical Information', 'clinical_information', NULL, 'Y', NULL),
  ('COMPOUND', 'Compound', NULL, NULL, 'Y', NULL),
  ('DISSTATE', 'Disease State / Stage / Tumor Grade', 'distease_state', NULL, 'Y', NULL),
  ('DOSE', 'Dose', 'dose', NULL, 'Y', NULL),
  ('GENMOD', 'Genetic Modification', NULL, NULL, 'Y', NULL),
  ('GENOTYPE', 'Genotype', 'genotype', NULL, 'Y', NULL),
  ('GROWTHCOND', 'Growth Conditions', NULL, NULL, 'Y', NULL),
  ('INDIV', 'Individual', 'individual', NULL, 'Y', NULL),
  ('ORGPART', 'Organ / Tissue', 'organism_part', NULL, 'Y', NULL),
  ('OTHER', 'Other', NULL, NULL, 'Y', NULL),
  ('SEX', 'Sex', 'sex', NULL, 'Y', NULL),
  ('TEMP', 'Temperature', NULL, NULL, 'Y', NULL),
  ('TIME', 'Time Course', 'sampling_time_point', NULL, 'Y', NULL);




INSERT INTO `gnomex`.`FundingAgency`(`idFundingAgency`, `fundingAgency`, `isPeerReviewedFunding`, `isActive`)
VALUES (4, 'NIH', 'Y', 'Y'),
  (5, 'Howard Hughes Foundation', 'Y', 'Y'),
  (6, 'American Cancer Society (ACS)\r\n', 'Y', 'Y'),
  (7, 'Centers for Disease Control and Prevention (CDC)\r\n', 'Y', 'Y'),
  (8, 'Food and Drug Administration (FDA)\r\n', 'Y', 'Y'),
  (9, 'National Science Foundation\r\n', 'Y', 'Y'),
  (10, 'US Army Department of Defense (DOD) (ovarian, breast, prostate programs)', 'Y', 'Y'),
  (11, 'Susan G. Komen Breast Cancer Foundation\r\n', 'Y', 'Y'),
  (12, 'Environmental Protection Agency', 'Y', 'Y'),
  (13, 'Leukemia and Lymphoma Society', 'Y', 'Y'),
  (14, 'Veterans Administration (central office, excluding local/regional/block grants)\r\n', 'Y', 'Y'),
  (15, 'Agency for Healthcare Research and Quality (AHRQ)\r\n', 'Y', 'Y'),
  (16, 'American Foundation for AIDS Research (AFAR)\r\n', 'Y', 'Y'),
  (17, 'American Institute for Cancer Research (AICR)\r\n', 'Y', 'Y'),
  (18, 'Cancer Research Foundation of America', 'Y', 'Y'),
  (19, 'Multiple Myeloma Research Foundation', 'Y', 'Y'),
  (20, 'National Institute for Occupational Safefy and Health (NIOSH)', 'Y', 'Y'),
  (21, 'University of California-Wide Breast Cancer Research Program', 'Y', 'Y'),
  (22, 'Other', 'N', 'Y');

INSERT INTO `gnomex`.`GenomeBuild`(`idGenomeBuild`, `genomeBuildName`, `idOrganism`, `isActive`, `isLatestBuild`, `idAppUser`)
VALUES (2, 'hg18; H_sapiens_Mar_2006; NCBI Build 36.1', 1, 'Y', 'Y', 566),
  (3, 'mm8, M_musculus_Mar_2006, NCBI Build 36', 3, 'Y', 'N', 566),
  (4, 'mm9, M_musculus_Jul_2007, NCBI Build 37', 3, 'Y', 'Y', 566),
  (5, 'ce6, C_elegans_May_2008, WS190', 6, 'Y', 'Y', 566),
  (6, 'Zv8, D_rerio_Jun_2008', 9, 'Y', 'Y', 566),
  (7, 'Zv7, D_rerio_Jul_2007', 9, 'Y', 'N', 566),
  (8, 'hg19; H_sapiens_Feb_2009; GRCh37', 1, 'Y', 'Y', 566),
  (9, 'TAIR8; A_thaliana_Mar_2008', 11, 'Y', 'Y', 273),
  (10, 'S_cerevisiae_20070113', 2, 'Y', 'Y', 273);



INSERT INTO `gnomex`.`Label`(`idLabel`, `label`, `isActive`)
VALUES (1, 'Cy3', 'Y'),
  (2, 'Cy5', 'Y');


INSERT INTO `gnomex`.`LabelingReactionSize`(`codeLabelingReactionSize`, `labelingReactionSize`, `isActive`, `sortOrder`)
VALUES ('STD', 'Standard', 'Y', 1),
  ('LRG', 'Large', 'Y', 2);

INSERT INTO `gnomex`.`Application`(`codeApplication`, `application`, `isActive`)
VALUES ('CGH', 'CGH', 'Y'),
  ('CHIP', 'ChiP-on-chip', 'Y'),
  ('EXON', 'Exon', 'Y'),
  ('EXP', 'Gene Expression', 'Y'),
  ('METH', 'DNA Methylation', 'Y'),
  ('MIRNA', 'miRNA', 'Y'),
  ('SNP', 'SNP', 'Y'),
  ('WTRANSCRP', 'HybMap', 'Y'),
  ('CHIPSEQ', 'ChIP-Seq', 'Y'),
  ('DMRNASEQ', 'Directional mRNA Sequencing', 'Y'),
  ('DNASEQ', 'Genomic DNA Sequencing', 'Y'),
  ('TDNASEQ', 'Targeted Genomic DNA Sequencing (capture/release)', 'Y'),
  ('MRNASEQ', 'mRNA Sequencing', 'Y'),
  ('SMRNASEQ', 'Small mRNA Sequencing', 'Y'),
  ('BIOAN', 'Bioanalyzer', 'Y'),
  ('QUBIT', 'Qubit picoGreen', 'Y');

INSERT INTO `gnomex`.`NumberSequencingCycles`(`idNumberSequencingCycles`, `numberSequencingCycles`, `isActive`)
VALUES (1, 18, 'N'),
  (2, 26, 'N'),
  (3, 36, 'Y'),
  (4, 76, 'Y');

INSERT INTO `gnomex`.`Organism`(`idOrganism`, `organism`, `abbreviation`, `mageOntologyCode`, `mageOntologyDefinition`, `isActive`, `idAppUser`)
VALUES (1, 'Human', NULL, 'HUMAN', NULL, 'Y', NULL),
  (2, 'Yeast', 'Y', NULL, NULL, 'Y', NULL),
  (3, 'Mouse', 'M', NULL, NULL, 'Y', NULL),
  (4, 'Drosophila', 'D', NULL, NULL, 'Y', NULL),
  (5, 'Planaria', 'P', NULL, NULL, 'Y', NULL),
  (6, 'C. elegans', 'Ce', NULL, NULL, 'Y', NULL),
  (7, 'Newt', 'N', NULL, NULL, 'Y', NULL),
  (8, 'Xenopus', 'X', NULL, NULL, 'Y', NULL),
  (9, 'Zebrafish', 'Z', NULL, NULL, 'Y', NULL),
  (10, 'Rat', 'R', NULL, NULL, 'Y', NULL),
  (11, 'Arabidopsis', 'A', NULL, NULL, 'Y', NULL),
  (12, 'Other (3rd party)', 'O', NULL, NULL, 'N', NULL),
  (13, 'Chicken', 'Ch', NULL, NULL, 'Y', NULL),
  (14, 'S. pombe', 'Sp', NULL, NULL, 'Y', NULL),
  (15, 'Canis familiaris', 'Cf', NULL, NULL, 'Y', NULL),
  (16, 'Emergen', 'E', NULL, NULL, 'N', NULL),
  (17, 'BioMicro', 'B', NULL, NULL, 'N', NULL),
  (18, 'Control', 'C', NULL, NULL, 'N', NULL),
  (19, 'Yeast Intergenic', 'Yi', NULL, NULL, 'N', NULL),
  (20, 'Unknown', '', NULL, NULL, 'N', NULL),
  (21, 'Sheep', NULL, NULL, NULL, 'Y', NULL),
  (22, 'Sodalis', NULL, NULL, NULL, 'Y', NULL),
  (23, 'Pig', 'Ss', NULL, NULL, 'Y', NULL),
  (24, 'Grape', 'vr', NULL, NULL, 'Y', NULL),
  (25, 'Asteromyia carbonifera', 'AC', NULL, NULL, 'Y', NULL),
  (26, 'Aspergillus fumigatus', 'Af', NULL, NULL, 'Y', NULL);

INSERT INTO `gnomex`.`Property` (`idProperty`,`propertyName`,`propertyValue`,`propertyDescription`, `forServerOnly`) VALUES 
 (1,'workauth_instructions','If billing account is inactive, please download the work authorization form below.<br> Forward the completed form to the [ENTER FACILITY NAME HERE] ([ENTER ROOM AND PHONE NUMBER HERE])','The instructions about work authorization that appear on the Experiment Submit Request page', 'N'),
 (2,'qc_instructions','Terms:  Bioanalyzer chips will be run when sufficient samples have been collected to load all wells of a chip.  Expedited service can be obtained if the researcher is willing to accept charges for any remaining empty wells on a chip.  Call [ENTER CORE FACILITY INFO HERE] for details.','The instructions regarding sample quality experiments.  These instructions about on Sample Tab of Experiment Submit Request window.', 'N'),
 (3,'workauth_pdf_url','doc/blank_microarray_core_work_authorization_form.pdf','The URL to the work authorization document (pdf).', 'N'),
 (4,'workauth_doc_url','doc/blank_microarray_core_work_authorization_form.doc','The URL to the work authoriztion document (word doc).', 'N'),
 (5,'contact_name_software_bugs','Bionformatics Core','Software org or contact', 'N'),
 (6,'contact_email_software_bugs','first.last@location.edu','Email address for reporting software bugs', 'N'),
 (7,'contact_name_core_facility','','Core facility\'s contact name', 'N'),
 (8,'contact_email_core_facility','first.last@somwhere.edu','Core facility\'s email adress', 'N'),
 (9,'help_url','http://www.hci.utah.edu/internal/CSG/PublicDocs/SourceDocs/HCIApps/GNomEx/GNomEx.htm','The URL to the GNomEx help documentation', 'N'),
 (10,'pricesheet1_name','Affymetrix Microarray Services','The name of the first price sheet, which shows as a link on window that appears when the \'Pricing\' link is pressed.', 'N'),
 (11,'pricesheet1_url','doc/blank_microarray_core_pricing_affy.doc','The URL to the first price sheet.', 'N'),
 (12,'pricesheet2_name','Agilent Microarray Services','(see description under pricesheet1_name)', 'N'),
 (13,'pricesheet2_url','doc/blank_microarray_core_pricing_agilent.doc','(see description under pricesheet1_url)', 'N'),
 (14,'pricesheet3_name','Sample Quality Services','(see description under pricesheet1_name)', 'N'),
 (15,'pricesheet3_url','doc/blank_microarray_core_pricing_sampleQuality.doc','(see description under pricesheet1_url)', 'N'),
 (16,'pricesheet4_name','Illumina Sequencing Services','(see description under pricesheet1_name)', 'N'),
 (17,'pricesheet4_url','doc/blank_microarray_core_pricing_illumina.doc','(see description under pricesheet1_url)', 'N'),
 (18,'core_facility_name','Microarray and Genomic Analysis Core Facility','The name of the core facility', 'N'),
 (19,'university_user_authentication','N','Always set to \'N\' for open source GNomEx.  Indicates that authentication is performed against local user name and passwords, stored in GNomEx database.', 'N'),
 (20,'generate_general_ledger_interface','N','Always set to \'N\' for open source GNomEx.  Indicates that GL interface file cannot be generated.', 'N'),
 (21,'contact_email_software_tester','softwaredeveloper.name@somewhere.edu','Email address of the software tester.  Test will receive auto emails (on test server) when this email is entered as user\'s email or group contact\'s email.', 'N'),
 (22,'contact_phone_core_facility','','The phone number of the core facility.', 'N'),
 (23,'flowcell_directory_flag','FC','Always set to \'FC\'.  Indicates the prefix of the flowcell directory name.', 'N'),
 (24,'qc_directory','bioanalysis','The name of the QC folder under the experiment folder.  Place all sample quality (bioanalysis, spectrophotometer reports, etc.) in this directory', 'N'),
 (25,'production_server','','The DNS name or IP address of the production app server, example:  myprodserver.hci.utah.edu', 'N'),
 (26,'lucene_index_directory','/home/orion/luceneIndexGnomEx','The file directory for storing lucene index files.', 'Y'),
 (27,'lucene_experiment_index_directory','/home/orion/luceneIndexGnomEx/Experiment','The file directory for storing lucene index files on experiment data.', 'Y'),
 (28,'lucene_analysis_index_directory','/home/orion/luceneIndexGnomEx/Analysis','The file directory for storing lucene index files on analysis data.', 'Y'),
 (29,'lucene_protocol_index_directory','/home/orion/luceneIndexGnomEx/Protocol','The file directory for storing lucene index files on protocol data.', 'Y'),
 (30,'experiment_directory','/home/gnomex/experimentData/','The file directory where experiment results are stored.', 'Y'),
 (31,'experiment_test_directory','/home/gnomex/experimentData/','The test file directory were experiment results are stored.', 'Y'),
 (32,'analysis_directory','/home/gnomex/analysisData/','The file directory where analysis files are stored.', 'Y'),
 (33,'analysis_test_directory','/home/gnomex/analysisData/','The test file directory where analysis files are stored.', 'Y'),
 (34,'flowcell_directory','/home/gnomex/flowCellData/','The file directory where sequencing pipeline results for a flowcell are stored.', 'Y'),
 (35,'flowcell_test_directory','/home/gnomex/flowCellData/','The test file directory where sequencing pipeline results for a flowcell are stored.', 'Y'),
 (36,'billing_core_facility_business_unit','','Always blank for open source GNomEx', 'Y'),
 (37,'billing_core_facility_account','','Always blank for open source GNomEx', 'Y'),
 (38,'billing_core_facility_fund','','Always blank for open source GNomEx', 'Y'),
 (39,'billing_core_facility_org','','Always blank for open source GNomEx', 'Y'),
 (40,'billing_core_facility_activity','','Always blank for open source GNomEx', 'Y'),
 (41,'temp_directory','/temp','The temp directory where temp files are created', 'Y'),
 (42,'illumina_barcoding_note','','Illumina sample barcoding instructions', 'N');



INSERT INTO `gnomex`.`ProtocolType`(`codeProtocolType`, `protocolType`, `isActive`)
VALUES ('IACUC', 'IACUC', 'Y'),
  ('IRB', 'IRB', 'Y');

INSERT INTO `gnomex`.`QualityControlStep`(`codeQualityControlStep`, `qualityControlStep`, `mageOntologyCode`, `mageOntologyDefinition`, `isActive`)
VALUES ('BIOREPL', 'Biological Replicate', 'biological_replicate', NULL, 'Y'),
  ('DYESWAP', 'Dye Swap', 'dye_swap_quality_control', NULL, 'Y'),
  ('NORTHBLOT', 'Northern Blotting', NULL, NULL, 'Y'),
  ('OTHER', 'Other Quality Control Step Taken (specify)', NULL, NULL, 'Y'),
  ('OTHERVALID', 'Other Validation (specify)', NULL, NULL, 'Y'),
  ('PCR', 'PCR', NULL, NULL, 'Y'),
  ('PEER', 'Peer Review', 'peer_review_quality_control', NULL, 'Y'),
  ('QUALPCR', 'Qualitative PCR', NULL, NULL, 'Y'),
  ('REVTPCR', 'Reverse Transcription PCR', 'reverse_transcription_PCR_quality_control', NULL, 'Y'),
  ('SPIKE', 'Spike', 'spike_quality_control', NULL, 'Y'),
  ('TECHREPL', 'Technical Replicate', 'technical_replicate', NULL, 'Y');


INSERT INTO `gnomex`.`RequestCategory`(`codeRequestCategory`, `requestCategory`, `idVendor`, `isActive`, `numberOfChannels`, `notes`)
VALUES ('AFFY', 'Affymetrix Microarray', 2, 'Y', 1, 'Gene expression, SNP analysis'),
  ('AGIL', 'Agilent 2-color Microarray', 1, 'Y', 2, 'Gene expression, CGH, ChIP-on-chip'),
  ('INHOUSE', 'In-house Spotted Microarray', 4, 'N', 2, null),
  ('NIMBLE', 'NimbleGen Microarray', 6, 'N', 2, null),
  ('OTHER', 'Microarray (Other)', NULL, 'N', NULL, null),
  ('QC', 'Sample Quality', NULL, 'Y', NULL, 'RNA NanoChip, RNA PicoChip, DNA 1000 chip, Qubit picoGreen, gDNA gel'),
  ('SOLEXA', 'Illumina Sequencing', 7, 'Y', NULL, 'Genomic seq, mRNA seq, directional mRNAseq, ChIP-seq, small RNA seq, targeted genomic seq (capture/release)'),
  ('AGIL1', 'Agilent 1-color Microarray', 1, 'Y', 1, 'Gene expression; miRNA');

INSERT INTO `gnomex`.`RequestCategoryApplication`(`codeRequestCategory`, `codeApplication`)
VALUES ('AFFY', 'CHIP'),
  ('AFFY', 'EXON'),
  ('AFFY', 'EXP'),
  ('AFFY', 'SNP'),
  ('AGIL', 'CGH'),
  ('AGIL', 'CHIP'),
  ('AGIL', 'EXP'),
  ('AGIL1', 'CGH'),
  ('AGIL1', 'CHIP'),
  ('AGIL1', 'EXP'),
  ('AGIL1', 'METH'),
  ('AGIL1', 'MIRNA'),
  ('AGIL1', 'WTRANSCRP'),
  ('INHOUSE', 'EXP'),
  ('NIMBLE', 'EXP'),
  ('NIMBLE', 'METH'),
  ('QC', 'BIOAN'),
  ('QC', 'QUBIT');


INSERT INTO `gnomex`.`SampleCharacteristic`(`codeSampleCharacteristic`, `sampleCharacteristic`, `mageOntologyCode`, `mageOntologyDefinition`, `isActive`, `idAppUser`)
VALUES ('AGE', 'Age / Developmental Stage', 'age', NULL, 'Y', NULL),
  ('CELLLINE', 'Cell Line / Strain', 'cell_line', NULL, 'Y', NULL),
  ('CELLTYPE', 'Cell Type', 'cell_type', NULL, 'Y', NULL),
  ('CLININFO', 'Clinical Information', 'clinical_information', NULL, 'Y', NULL),
  ('COMPOUND', 'Compound', NULL, NULL, 'Y', NULL),
  ('DISSTATE', 'Disease State / Stage / Tumor Grade', 'distease_state', NULL, 'Y', NULL),
  ('DOSE', 'Dose', 'dose', NULL, 'Y', NULL),
  ('GENMOD', 'Genetic Modification', NULL, NULL, 'Y', NULL),
  ('GENOTYPE', 'Genotype', 'genotype', NULL, 'Y', NULL),
  ('GROWTHCOND', 'Growth Conditions', NULL, NULL, 'Y', NULL),
  ('INDIV', 'Individual', 'individual', NULL, 'Y', NULL),
  ('ORGPART', 'Organ / Tissue', 'organism_part', NULL, 'Y', NULL),
  ('OTHER', 'Other', NULL, NULL, 'Y', NULL),
  ('SEX', 'Sex', 'sex', NULL, 'Y', NULL),
  ('TEMP', 'Temperature', NULL, NULL, 'Y', NULL),
  ('TIME', 'Time Course', 'sampling_time_point', NULL, 'Y', NULL);

INSERT INTO `gnomex`.`SamplePrepMethod`(`idSamplePrepMethod`, `samplePrepMethod`, `isActive`)
VALUES (1, 'Qiagen RNeasy mini kit', 'Y'),
  (2, 'Trizol Reagent', 'Y'),
  (3, 'Agilent total RNA isolation mini kit', 'Y'),
  (4, 'RiboAmp', 'N'),
  (5, 'MessageAmp', 'N'),
  (6, 'Phenol/chloroform', 'Y'),
  (20, 'Gentra Puregene DNA Purification Kit', 'Y'),
  (21, 'Qiagen DNeasy tissue kit', 'Y'),
  (22, 'Other', 'Y'),
  (23, 'Agilent plant RNA isolation mini kit', 'Y'),
  (24, 'Ambion Tri Reagent', 'Y'),
  (25, 'Ambion RNAqueous kit', 'Y'),
  (26, 'Promega SV Total RNA Isolation kit', 'Y'),
  (27, 'Qiagen RNeasy Plant kit', 'Y'),
  (28, 'Ambion mirVana miRNA isolation kit', 'Y'),
  (29, 'Qiagen RNeasy Micro Kit', 'Y'),
  (30, 'Qiagen Genomic-tip', 'Y'),
  (31, 'Illumina Genomic DNA Sample Prep Kit ', 'Y'),
  (32, 'Illumina DGE-Tag Profiling NlaIII Sample Prep Kit', 'N'),
  (33, 'Illumina DGE-Tag Profiling DpnII Sample Prep Kit', 'N'),
  (34, 'Illumina ChIP-Seq prep kit', 'Y'),
  (35, 'Illumina Small RNA Sample Prep Kit ', 'Y'),
  (39, 'Illumina mRNA-Seq Sample Prep Kit', 'Y'),
  (41, 'Qiagen miRNeasy Mini Kit', 'Y');

INSERT INTO `gnomex`.`SamplePrepMethodRequestCategory`(`idSamplePrepMethodRequestCategory`, `idSamplePrepMethod`, `codeRequestCategory`)
VALUES (1, 1, 'AGIL'),
  (2, 2, 'AGIL'),
  (3, 3, 'AGIL'),
  (4, 4, 'AGIL'),
  (5, 5, 'AGIL'),
  (6, 6, 'AGIL'),
  (7, 20, 'AGIL'),
  (8, 21, 'AGIL'),
  (9, 22, 'AGIL'),
  (10, 23, 'AGIL'),
  (11, 24, 'AGIL'),
  (12, 25, 'AGIL'),
  (13, 26, 'AGIL'),
  (14, 27, 'AGIL'),
  (15, 28, 'AGIL'),
  (16, 29, 'AGIL'),
  (17, 30, 'AGIL'),
  (18, 1, 'AFFY'),
  (19, 2, 'AFFY'),
  (20, 3, 'AFFY'),
  (21, 4, 'AFFY'),
  (22, 5, 'AFFY'),
  (23, 6, 'AFFY'),
  (24, 20, 'AFFY'),
  (25, 21, 'AFFY'),
  (26, 22, 'AFFY'),
  (27, 23, 'AFFY'),
  (28, 24, 'AFFY'),
  (29, 25, 'AFFY'),
  (30, 26, 'AFFY'),
  (31, 27, 'AFFY'),
  (32, 28, 'AFFY'),
  (33, 29, 'AFFY'),
  (34, 30, 'AFFY'),
  (35, 1, 'QC'),
  (36, 2, 'QC'),
  (37, 3, 'QC'),
  (38, 4, 'QC'),
  (39, 5, 'QC'),
  (40, 6, 'QC'),
  (41, 20, 'QC'),
  (42, 21, 'QC'),
  (43, 22, 'QC'),
  (44, 23, 'QC'),
  (45, 24, 'QC'),
  (46, 25, 'QC'),
  (47, 26, 'QC'),
  (48, 27, 'QC'),
  (49, 28, 'QC'),
  (50, 29, 'QC'),
  (51, 30, 'QC'),
  (52, 31, 'SOLEXA'),
  (53, 32, 'SOLEXA'),
  (54, 33, 'SOLEXA'),
  (55, 34, 'SOLEXA'),
  (56, 35, 'SOLEXA'),
  (57, 39, 'SOLEXA'),
  (58, 1, 'AGIL1'),
  (59, 2, 'AGIL1'),
  (60, 3, 'AGIL1'),
  (61, 4, 'AGIL1'),
  (62, 5, 'AGIL1'),
  (63, 6, 'AGIL1'),
  (64, 20, 'AGIL1'),
  (65, 21, 'AGIL1'),
  (66, 22, 'AGIL1'),
  (67, 23, 'AGIL1'),
  (68, 24, 'AGIL1'),
  (69, 25, 'AGIL1'),
  (70, 26, 'AGIL1'),
  (71, 27, 'AGIL1'),
  (72, 28, 'AGIL1'),
  (73, 29, 'AGIL1'),
  (74, 30, 'AGIL1');

INSERT INTO `gnomex`.`SamplePrepMethodSampleType`(`idSamplePrepMethodSampleType`, `idSampleType`, `idSamplePrepMethod`, `isDefaultForSampleType`, `isActive`)
VALUES (2, 5, 23, 'N', 'Y'),
  (3, 5, 3, 'N', 'Y'),
  (4, 7, 20, 'N', 'Y'),
  (5, 1, 20, 'Y', 'Y'),
  (6, 6, 20, 'Y', 'Y'),
  (9, 1, 6, 'N', 'Y'),
  (11, 5, 6, 'N', 'Y'),
  (13, 1, 21, 'N', 'Y'),
  (15, 5, 1, 'Y', 'Y'),
  (19, 1, 2, 'N', 'Y'),
  (21, 5, 2, 'N', 'Y'),
  (22, 1, 22, 'N', 'Y'),
  (23, 6, 22, 'N', 'Y'),
  (24, 5, 22, 'N', 'Y'),
  (25, 2, 22, 'N', 'Y'),
  (26, 7, 22, 'N', 'Y'),
  (27, 4, 22, 'N', 'Y'),
  (28, 3, 22, 'N', 'Y'),
  (29, 8, 22, 'N', 'Y'),
  (30, 5, 29, NULL, 'Y'),
  (31, 1, 30, NULL, 'Y'),
  (32, 7, 34, 'Y', 'Y'),
  (33, 5, 33, 'N', 'N'),
  (34, 9, 33, 'N', 'N'),
  (35, 5, 32, 'N', 'N'),
  (36, 9, 32, 'N', 'N'),
  (37, 1, 31, 'Y', 'Y'),
  (38, 12, 35, 'Y', 'Y'),
  (39, 5, 39, 'Y', 'Y'),
  (40, 4, 39, 'Y', 'Y'),
  (41, 9, 39, 'Y', 'Y');

INSERT INTO `gnomex`.`SamplePrepMethodSeqRunType`(`idSamplePrepMethodSeqRunType`, `idSamplePrepMethod`, `idSeqRunType`)
VALUES (1, 34, 3),
  (2, 33, 3),
  (3, 32, 3),
  (4, 31, 3),
  (5, 35, 3),
  (6, 34, 4),
  (7, 31, 4),
  (8, 39, 4),
  (9, 39, 3);

INSERT INTO `gnomex`.`SampleSource`(`idSampleSource`, `sampleSource`, `isActive`)
VALUES (1, 'Cell Line', 'Y'),
  (2, 'Organ', 'Y'),
  (3, 'Whole Organism', 'Y'),
  (4, 'Tissue', 'Y'),
  (5, 'Other (specify in notes)', 'Y'),
  (6, 'Blood', 'Y');

INSERT INTO `gnomex`.`SampleType`(`idSampleType`, `sampleType`, `sortOrder`, `isActive`)
VALUES (1, 'DNA', 3, 'Y'),
  (2, 'aRNA', 6, 'N'),
  (3, 'miRNA', 8, 'Y'),
  (4, 'mRNA', 7, 'Y'),
  (5, 'Total RNA (eukaryote)', 1, 'Y'),
  (6, 'FFPE DNA', 5, 'Y'),
  (7, 'chIP DNA', 4, 'Y'),
  (8, 'polyA RNA', NULL, 'N'),
  (9, 'Total RNA (prokaryote)', 2, 'Y'),
  (10, 'cDNA', 5, 'Y'),
  (11, 'cap RNA', 6, 'Y'),
  (12, 'small RNA', 10, 'Y');

INSERT INTO `gnomex`.`SampleTypeApplication`(`idSampleTypeApplication`, `idSampleType`, `codeApplication`, `idLabelingProtocolDefault`, `idHybProtocolDefault`, `idScanProtocolDefault`, `idFeatureExtractionProtocolDefault`, `isActive`)
VALUES (2, 5, 'EXP', null, null, null, null, 'Y'),
  (3, 4, 'EXP', NULL, NULL, NULL, NULL, 'Y'),
  (4, 3, 'EXP', NULL, NULL, NULL, NULL, 'Y'),
  (5, 2, 'EXP', NULL, NULL, NULL, NULL, 'N'),
  (6, 1, 'CGH', null, null, null, null, 'Y'),
  (7, 6, 'CGH', NULL, NULL, NULL, NULL, 'Y'),
  (8, 7, 'CHIP', null, null, null, null, 'Y'),
  (9, 1, 'METH', null, null, null, null, 'Y'),
  (10, 7, 'METH', null, null, null, null, 'Y'),
  (11, 1, 'SNP', null, null, null, null, 'Y'),
  (12, 5, 'WTRANSCRP', null, null, null, null, 'Y'),
  (13, 5, 'MIRNA', null, null, null, null, 'Y'),
  (14, 7, 'CHIPSEQ', null, null, null, null, 'Y'),
  (15, 4, 'DMRNASEQ', null, null, null, null, 'Y'),
  (16, 1, 'DNASEQ', null, null, null, null, 'Y'),
  (17, 1, 'TDNASEQ', null, null, null, null, 'Y'),
  (18, 4, 'MRNASEQ', null, null, null, null, 'Y'),
  (19, 12, 'SMRNASEQ', null, null, null, null, 'Y');


INSERT INTO `gnomex`.`SampleTypeRequestCategory`(`idSampleTypeRequestCategory`, `idSampleType`, `codeRequestCategory`)
VALUES (1, 1, 'AGIL'),
  (2, 2, 'AGIL'),
  (3, 3, 'AGIL'),
  (4, 4, 'AGIL'),
  (5, 5, 'AGIL'),
  (6, 6, 'AGIL'),
  (7, 7, 'AGIL'),
  (8, 8, 'AGIL'),
  (9, 9, 'AGIL'),
  (10, 1, 'QC'),
  (11, 2, 'QC'),
  (12, 3, 'QC'),
  (13, 4, 'QC'),
  (14, 5, 'QC'),
  (15, 6, 'QC'),
  (16, 7, 'QC'),
  (17, 8, 'QC'),
  (18, 9, 'QC'),
  (19, 1, 'AFFY'),
  (20, 2, 'AFFY'),
  (21, 3, 'AFFY'),
  (22, 4, 'AFFY'),
  (23, 5, 'AFFY'),
  (24, 6, 'AFFY'),
  (25, 7, 'AFFY'),
  (26, 8, 'AFFY'),
  (27, 9, 'AFFY'),
  (28, 1, 'SOLEXA'),
  (29, 5, 'SOLEXA'),
  (30, 9, 'SOLEXA'),
  (31, 7, 'SOLEXA'),
  (32, 4, 'SOLEXA'),
  (33, 9, 'SOLEXA'),
  (34, 10, 'SOLEXA'),
  (35, 11, 'SOLEXA'),
  (36, 12, 'SOLEXA'),
  (37, 2, 'AGIL1'),
  (38, 3, 'AGIL1'),
  (39, 4, 'AGIL1'),
  (40, 5, 'AGIL1'),
  (41, 6, 'AGIL1'),
  (42, 7, 'AGIL1'),
  (43, 8, 'AGIL1'),
  (44, 9, 'AGIL1');



INSERT INTO `gnomex`.`SeqRunType`(`idSeqRunType`, `seqRunType`, `isActive`)
VALUES (1, 'Sequencing / chIP Sequencing', 'N'),
  (2, 'Digital Gene Expression', 'N'),
  (3, 'Single-end reads', 'Y'),
  (4, 'Paired-end reads', 'Y');



INSERT INTO `gnomex`.`SequencingControl`(`idSequencingControl`, `sequencingControl`, `isActive`, `idAppUser`)
VALUES (1, 'PhiX Control', 'Y', NULL);

INSERT INTO `gnomex`.`SlideDesign`(`idSlideDesign`, `name`, `slideDesignProtocolName`, `idSlideProduct`, `accessionNumberArrayExpress`, `isActive`)
VALUES (3,'Whole human genome microarray (44K)','012391',2,'A-AGIL-28','N'),
 (4,'Human promoter 1 ChIP-on-chip (44K)',NULL,3,NULL,'N'),
 (5,'Human promoter 2 ChIP-on-chip (44K)',NULL,3,NULL,'Y'),
 (6,'Human 1A GE microarray (22K)',NULL,5,NULL,'N'),
 (8,'Human genome CGH 44B (4x44K)','013282',7,NULL,'Y'),
 (11,'Human Genome CGH (1x185K)',NULL,10,NULL,'N'),
 (12,'Arabidopsis 3 oligo microarray (44K)',NULL,11,'A-AGIL-12','N'),
 (14,'C. elegans Gene Expression (22K)','012795',13,NULL,'N'),
 (15,'Canine Gene Expression v2 (4x44K)','021193',14,NULL,'Y'),
 (17,'Chicken Gene Expression (4x44k)','015068',16,NULL,'Y'),
 (18,'Whole mouse genome microarray (44K)',NULL,17,NULL,'N'),
 (25,'Rat oligo microarray (22K)',NULL,24,NULL,'N'),
 (26,'Whole rat genome microarray (44K)','014879',25,NULL,'Y'),
 (31,'Xenopus oligo expression microarray (22K)','013665',30,NULL,'N'),
 (32,'Yeast V2 oligo expression microarray (11K)','013884',31,NULL,'N'),
 (33,'S. cerevisiae ChIP array(4x44K)','014810',32,NULL,'Y'),
 (36,'Zebrafish oligo expression microarray (22K)',NULL,35,'A-AGIL-ZEBRAFISH','N'),
 (54,'Whole human genome microarray (4x44K)','014850',54,'A-AGIL-28','Y'),
 (114,'Yeast Whole Genome ChIP-on-chip microarray (244K)','14741',110,NULL,'Y'),
 (115,'Human Genome CGH 244A (1x244K)','014693',111,NULL,'Y'),
 (117,'Human Genome CGH 105A (2x105K)','014698',113,NULL,'Y'),
 (118,'Affymetrix GeneChip Human Genome U133 Plus 2.0',NULL,114,NULL,'Y'),
 (126,'Human GeneChip Mapping 250K Nsp Array',NULL,123,NULL,'Y'),
 (128,'Arabidopsis Gene Expression v3 (4x44K)','015059',124,NULL,'Y'),
 (129,'Arabidopsis Whole Genome ChIP-on-chip slide 1','014798',125,NULL,'Y'),
 (130,'Arabidopsis Whole Genome ChIP-on-chip Slide 2','014799',125,NULL,'Y'),
 (131,'C. elegans Gene Expression (4x44K)','015061',126,NULL,'Y'),
 (132,'Zebrafish oligo microarray (4x44K)','015064',127,NULL,'N'),
 (133,'Xenopus oligo microarray (4x44K)','015066',128,NULL,'Y'),
 (135,'Sodalis tiling microarray (4x44K)','015651',130,NULL,'Y'),
 (136,'Yeast GE microarray (4x44K)','015072',131,NULL,'Y'),
 (137,'Human Promoter ChIP-on-chip slide 1 (1x244K)','014706',132,NULL,'Y'),
 (138,'Human Promoter ChIP-on-chip slide 2 (1x244K)','014707',132,NULL,'Y'),
 (139,'Whole Mouse Genome Microarray (4x44K)','014868',133,NULL,'Y'),
 (142,'Canine SNP Array (520431)','520431',135,NULL,'Y'),
 (143,'Mouse Genome 430 2.0 Array','900496',136,NULL,'Y'),
 (144,'Human CpG Island (1x244K)','014791',137,NULL,'Y'),
 (148,'Mouse Promoter 1 ChIP-on-Chip (244K)','014716',141,NULL,'Y'),
 (149,'Mouse Promoter 2 ChIP-on-Chip (244K)','014717',141,NULL,'Y'),
 (151,'Human ENCODE ChIP-on-chip (1x244K)','014792',143,NULL,'Y'),
 (153,'Drosophila Gene Expression (4x44K)','018972',145,NULL,'Y'),
 (155,'Affymetrix Human SNP 5.0 Array',NULL,147,NULL,'Y'),
 (164,'Zebrafish Gene Expression v2 (4x44K)','19161',155,NULL,'Y'),
 (165,'Mouse miRNA microarray (8x15K)','019119',156,NULL,'Y'),
 (167,'Human miRNA v2 (8x15K)','016436',158,NULL,'Y'),
 (168,'Human CGH (4x180K)','022060',159,NULL,'Y'),
 (170,'Affymetrix Human SNP 6.0 Array',NULL,161,NULL,'Y'),
 (175,'Porcine Gene Expression Microarray','020109',166,NULL,'Y'),
 (177,'Mouse miRNA microarray v2 (8x15K)','021828',168,NULL,'Y'),
 (179,'Human miRNA v3 (8x15K)','021827',170,NULL,'Y');


INSERT INTO `gnomex`.`SlideProduct`(`idSlideProduct`, `name`, `catalogNumber`, `isCustom`, `idLab`, `codeApplication`, `idVendor`, `idOrganism`, `arraysPerSlide`, `slidesInSet`, `isSlideSet`, `isActive`, `idBillingSlideProductClass`, `idBillingSlideServiceClass`)
VALUES  (2,'Whole human genome expression array (44K)',NULL,'N',NULL,'EXP',1,1,1,1,'N','N',NULL,NULL),
 (3,'Human promoter ChIP-on-chip Set (2 slides) (44K)',NULL,'N',NULL,'CHIP',1,1,1,2,'Y','N',3,8),
 (5,'Human 1A GE microarray (22K)',NULL,'N',NULL,'EXP',1,1,1,1,'N','N',NULL,NULL),
 (7,'Human genome CGH 44B (4x44K)',NULL,'N',NULL,'CGH',1,1,1,1,'N','Y',3,4),
 (10,'Human Genome CGH (1x185K)',NULL,'N',NULL,'CGH',1,1,1,1,'N','N',NULL,NULL),
 (11,'Arabidopsis 3 oligo expression array (44K)',NULL,'N',NULL,'EXP',1,11,1,1,'N','N',NULL,NULL),
 (13,'C. elegans Gene Expression (22K)',NULL,'N',NULL,'EXP',1,6,1,1,'N','N',NULL,NULL),
 (14,'Canine Gene Expression v2 (4x44K)',NULL,'N',NULL,'EXP',1,15,4,1,'N','Y',3,1),
 (16,'Chicken Gene Expression (4x44k)',NULL,'N',NULL,'EXP',1,13,4,1,'N','Y',3,1),
 (17,'Whole mouse genome expression array (44K)',NULL,'N',NULL,'EXP',1,3,1,1,'N','N',NULL,NULL),
 (24,'Rat oligo expression array (22K)',NULL,'N',NULL,'EXP',1,10,1,1,'N','N',NULL,NULL),
 (25,'Whole rat genome expression array (44K)',NULL,'N',NULL,'EXP',1,10,4,1,'N','Y',3,1),
 (30,'Xenopus oligo expression microarray (22K)',NULL,'N',NULL,'EXP',1,8,1,1,'N','N',NULL,NULL),
 (31,'S. cerevisiae V2 oligo expression array (11K)',NULL,'N',NULL,'EXP',1,2,2,1,'N','N',NULL,NULL),
 (32,'S. cerevisiae ChIP array(4x44K)',NULL,'N',NULL,'CHIP',1,2,4,1,'N','Y',3,8),
 (35,'Zebrafish oligo expression array (22K)',NULL,'N',NULL,'EXP',1,9,1,1,'N','N',NULL,NULL),
 (54,'Whole human genome expression array (44K) (4 pack)',NULL,'N',NULL,'EXP',1,1,4,1,'N','Y',3,1),
 (110,'Yeast Whole Genome ChIP-on-chip microarray (244K)',NULL,'N',NULL,'CHIP',1,2,1,1,'N','Y',1,6),
 (111,'Human Genome CGH 244A (1x244K)',NULL,'N',NULL,'CGH',1,1,1,1,'N','Y',1,2),
 (113,'Human Genome CGH 105A (2x105K)',NULL,'N',NULL,'CGH',1,1,2,1,'N','Y',2,3),
 (114,'Affymetrix GeneChip Human Genome U133 Plus 2.0',NULL,'N',NULL,'EXP',2,1,1,1,'N','Y',7,12),
 (123,'Human GeneChip Mapping 250K Nsp Array',NULL,'N',NULL,'SNP',2,1,1,1,'N','Y',5,13),
 (124,'Arabidopsis Gene Expression v3 (4x44K)',NULL,'N',NULL,'EXP',1,11,4,1,'N','Y',3,1),
 (125,'Arabidopsis Whole Genome ChIP-on-chip Set (244K)',NULL,'N',NULL,'CHIP',1,11,1,2,'Y','Y',1,6),
 (126,'C. elegans Gene Expression (4x44K)',NULL,'N',NULL,'EXP',1,6,4,1,'N','Y',3,1),
 (127,'Zebrafish oligo microarray (4x44K)',NULL,'N',NULL,'EXP',1,9,4,1,'N','N',3,1),
 (128,'Xenopus oligo microarray (4x44K)',NULL,'N',NULL,'EXP',1,8,4,1,'N','Y',3,1),
 (130,'Sodalis tiling microarray (4x44K)',NULL,'N',NULL,'EXP',1,22,4,1,'N','Y',NULL,NULL),
 (131,'Yeast GE microarray (4x44K)',NULL,'N',NULL,'EXP',1,2,4,1,'N','Y',3,1),
 (132,'Human Promoter ChIP-on-chip 244K Set',NULL,'N',NULL,'CHIP',1,1,1,2,'Y','Y',1,6),
 (133,'Whole Mouse Genome Microarray (4x44K)',NULL,'N',NULL,'EXP',1,3,4,1,'N','Y',3,1),
 (135,'Canine SNP Array (520431)',NULL,'N',NULL,NULL,NULL,NULL,1,1,'N','Y',5,13),
 (136,'Mouse Genome 430 2.0 Array',NULL,'N',NULL,NULL,2,3,1,1,'N','Y',7,12),
 (137,'Human CpG Island (1x244K)',NULL,'N',NULL,NULL,1,1,1,1,'N','Y',1,6),
 (141,'Mouse Promoter ChIP-on-chip Set (244K)',NULL,'N',NULL,NULL,1,3,1,2,'Y','Y',1,6),
 (143,'Human ENCODE ChIP-on-chip (1x244K)',NULL,'N',NULL,NULL,1,1,1,1,'N','Y',1,6),
 (145,'Drosophila Gene Expression (4x44K)',NULL,'N',NULL,NULL,1,4,4,1,'N','Y',3,1),
 (147,'Affymetrix Human SNP 5.0 Array','901069','N',NULL,NULL,2,1,1,1,'N','Y',6,13),
 (155,'Zebrafish Gene Expression v2 (4x44K)',NULL,'N',NULL,NULL,1,9,4,1,'N','Y',3,1),
 (156,'Mouse miRNA microarray (8x15K)',NULL,'N',NULL,NULL,1,3,8,1,'N','Y',11,16),
 (158,'Human miRNA v2 (8x15K)',NULL,'N',NULL,NULL,1,1,8,1,'N','Y',11,16),
 (159,'Human CGH (4x180K)',NULL,'N',NULL,NULL,1,1,4,1,'N','Y',13,19),
 (161,'Affymetrix Human SNP 6.0 Array',NULL,'N',NULL,NULL,2,1,1,1,'N','Y',8,14),
 (166,'Porcine Gene Expression Microarray',NULL,'N',NULL,NULL,1,23,4,1,'N','Y',3,1),
 (168,'Mouse miRNA microarray v2 (8x15K)',NULL,'N',NULL,NULL,1,3,8,1,'N','Y',11,16),
 (170,'Human miRNA v3 (8x15K)',NULL,'N',NULL,NULL,1,1,1,1,'N','Y',11,16);


INSERT INTO `gnomex`.`SlideProductApplication`(`idSlideProduct`, `codeApplication`)
VALUES  (7,'CGH'),
 (10,'CGH'),
 (111,'CGH'),
 (113,'CGH'),
 (159,'CGH'),
 (3,'CHIP'),
 (32,'CHIP'),
 (110,'CHIP'),
 (125,'CHIP'),
 (132,'CHIP'),
 (137,'CHIP'),
 (141,'CHIP'),
 (143,'CHIP'),
 (2,'EXP'),
 (5,'EXP'),
 (11,'EXP'),
 (13,'EXP'),
 (14,'EXP'),
 (16,'EXP'),
 (17,'EXP'),
 (24,'EXP'),
 (25,'EXP'),
 (30,'EXP'),
 (31,'EXP'),
 (35,'EXP'),
 (54,'EXP'),
 (114,'EXP'),
 (124,'EXP'),
 (126,'EXP'),
 (127,'EXP'),
 (128,'EXP'),
 (130,'EXP'),
 (131,'EXP'),
 (133,'EXP'),
 (136,'EXP'),
 (145,'EXP'),
 (155,'EXP'),
 (166,'EXP'),
 (132,'METH'),
 (137,'METH'),
 (156,'MIRNA'),
 (158,'MIRNA'),
 (168,'MIRNA'),
 (170,'MIRNA'),
 (123,'SNP'),
 (135,'SNP'),
 (147,'SNP'),
 (161,'SNP'),
 (143,'WTRANSCRP');


INSERT INTO `gnomex`.`SlideSource`(`codeSlideSource`, `slideSource`, `isActive`)
VALUES ('CLIENT', 'Lab', 'Y'),
  ('CORE', 'Core Facility', 'Y'),
  ('REUSE', 'Stripped Slide', 'Y');

INSERT INTO `gnomex`.`State`(`codeState`, `state`, `isActive`)
VALUES ('AK', 'Alaska', 'Y'),
  ('AL', 'Alabama', 'Y'),
  ('AR', 'Arkansas', 'Y'),
  ('AZ', 'Arizona', 'Y'),
  ('CA', 'California', 'Y'),
  ('CO', 'Colorado', 'Y'),
  ('CT', 'Conneticut', 'Y'),
  ('DE', 'Deleware', 'Y'),
  ('FL', 'Florida', 'Y'),
  ('GA', 'Georgia', 'Y'),
  ('HI', 'Hawaii', 'Y'),
  ('IA', 'Iowa', 'Y'),
  ('ID', 'Idaho', 'Y'),
  ('IL', 'Illinois', 'Y'),
  ('IN', 'Indiana', 'Y'),
  ('KS', 'Kansas', 'Y'),
  ('KY', 'Kentucky', 'Y'),
  ('LA', 'Lousiana', 'Y'),
  ('MA', 'Massachusetts', 'Y'),
  ('MD', 'Maryland', 'Y'),
  ('ME', 'Maine', 'Y'),
  ('MI', 'Michigan', 'Y'),
  ('MN', 'Minnesota', 'Y'),
  ('MO', 'Missouri', 'Y'),
  ('MS', 'Mississippi', 'Y'),
  ('MT', 'Montana', 'Y'),
  ('NC', 'North Carolina', 'Y'),
  ('ND', 'North Dakota', 'Y'),
  ('NE', 'Nebraska', 'Y'),
  ('NH', 'New Hampshire', 'Y'),
  ('NJ', 'New Jersey', 'Y'),
  ('NM', 'New Mexico', 'Y'),
  ('NV', 'Nevada', 'Y'),
  ('NY', 'New York', 'Y'),
  ('OH', 'Ohio', 'Y'),
  ('OK', 'Oklahoma', 'Y'),
  ('OR', 'Oregon', 'Y'),
  ('PA', 'Pennsylvania', 'Y'),
  ('RI', 'Rhode Island', 'Y'),
  ('SC', 'South Carolina', 'Y'),
  ('SD', 'South Dakota', 'Y'),
  ('TN', 'Tennessee', 'Y'),
  ('TX', 'Texas', 'Y'),
  ('UT', 'Utah', 'Y'),
  ('VA', 'Virginia', 'Y'),
  ('VT', 'Vermont', 'Y'),
  ('WA', 'Washington', 'Y'),
  ('WI', 'Wisconsin', 'Y'),
  ('WV', 'West Virginia', 'Y'),
  ('WY', 'Wyoming', 'Y');

INSERT INTO `gnomex`.`Step`(`codeStep`, `step`, `isActive`)
VALUES ('COMPLETE', 'Complete', 'Y'),
  ('EXT', 'Scan/Feature Extraction', 'Y'),
  ('HYB', 'Hybridization', 'Y'),
  ('LABEL', 'Labeling', 'Y'),
  ('QC', 'Quality Control', 'Y'),
  ('SEQASSEM', 'Solexa Cluster Gen', 'Y'),
  ('SEQPIPE', 'Illumina Data Pipeline', 'Y'),
  ('SEQPREP', 'Illumina Library Prep', 'Y'),
  ('SEQQC', 'Illumina Quality Control', 'Y'),
  ('SEQRUN', 'Illumina Sequencing Run', 'Y'),
  ('SEQSTOCK', 'Illumina Flow Cell Stock', 'Y'),
  ('STOP', 'Stop/Failed', 'Y');


INSERT INTO `gnomex`.`UserPermissionKind`(`codeUserPermissionKind`, `userPermissionKind`, `isActive`)
VALUES ('ADMIN', 'Admin', 'Y'),
  ('BILLING', 'Billing', 'Y'),
  ('LAB', 'Lab', 'Y'),
  ('USER', 'User', 'Y');

INSERT INTO `gnomex`.`Vendor`(`idVendor`, `vendorName`, `description`, `isActive`)
VALUES (1, 'Agilent', 'Agilent', 'Y'),
  (2, 'Affymetrix', 'Affymetrix', 'Y'),
  (4, 'In House', 'In House', 'N'),
  (5, 'Invitrogen', 'Invitrogen', 'N'),
  (6, 'NimbleGen', 'NimbleGen', 'N'),
  (7, 'Illumina', 'Illumina', 'Y');

INSERT INTO `gnomex`.`Visibility`(`codeVisibility`, `visibility`)
VALUES ('MEM', 'Members'),
  ('MEMCOL', 'Members and Collaborators'),
  ('PUBLIC', 'Public');

-- Re-enable foreign key checks
SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS;

-- End of script